### ISSUE TITLE
CopyItem in UnicornDataProvider fails

### ISSUE THREAD

**Issue by sbouwmeester** [kamsar/Unicorn#116]
Hi,

Since last week we upgraded to Unicorn 3.1.2. Currently we have 9 different configuration and they all use transparent sync

We have some an issue with the copyitem logic in the UnicornDataProvider. In our project we have some logic with copies some item to a specific location. This logic did result in an "Object reference not set to an instance of an object " exception..

After debugging the Unicorn project I figured out the following looks wrong:
- Resolving the destination item did not work. After changing this logic into the same as resolving the existing item (with useCache = true) the logic did work.
- The assert for the destination item checks the existingitem instead of the destination item.

Like this:
```
public virtual void CopyItem(ItemDefinition source, ItemDefinition destination, string copyName, ID copyId, CallContext context)
{
   if (DisableSerialization) return;
    
   // copying is easy - all we have to do is serialize the item using the copyID and new target destination as the parent. 
   // Copied children will all result in multiple calls to CopyItem so we don't even need to worry about them.
   var existingItem = GetSourceFromId(source.ID, true);
    
   Assert.IsNotNull(existingItem, "Existing item to copy was not in the database!");
    
   var destinationItem = GetSourceFromId(destination.ID, true);

   Assert.IsNotNull(destinationItem, "Copy destination was not in the database!");
    
  // wrap the existing item in a proxy so we can mutate it into a copy
  var copyTargetItem = new ProxyItem(existingItem);

  copyTargetItem.Path = $"{destinationItem.Path}/{copyName}";

```

Greetings,
Sander

**Comment by kamsar** [comment]
Tested your patch in a few ways and all seems well. It'll be in the next release. Thanks!

**Comment by kamsar** [comment]
Fixed in 3.1.4
